src/Community.VisualStudio.Toolkit.Shared/Helpers/OutputWindowPane.cs

```diff
@@ -0,0 +1,276 @@
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// Guid myPaneGuid;
+    /// {
+    ///     OutputWindowPane pane = await VS.Windows.CreateOutputWindowPaneAsync("My Pane");
+    ///     myPaneGuid = pane.Guid;
+    ///     await pane.WriteLineAsync("My message");
+    /// }
+    /// 
+    /// // Elsewhere:
+    /// {
+    ///     OutputWindowPane pane = await VS.Windows.GetOutputWindowPaneAsync(myPaneGuid);
+    ///     pane.Name = "My Pane - Updated";
+    ///     using (TextWriter writer = await pane.CreateOutputPaneTextWriterAsync())
+    ///     {
+    ///         char[] buffer = GetSomeChars();
+    ///         await writer.WriteLineAsync("This is a more efficient way to write lots of text.");
+    ///         await writer.WriteAsync(buffer, 0, buffer.Length);
+    ///         await writer.WriteLineAsync();
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    public class OutputWindowPane
+    {
+        private readonly string _newPaneName;
+        private IVsOutputWindowPane? _pane;
+
+        /// <summary>
+        /// Creates a new Output window pane with the given name (title).
+        /// The new pane can be created at construction time or lazily upon first write.
+        /// </summary>
+        /// <param name="name">The name (title) of the new pane.</param>
+        /// <param name="lazyCreate">Whether to lazily create the pane upon first write.</param>
+        /// <returns>A new OutputWindowPane.</returns>
+        public static async Task<OutputWindowPane> CreateAsync(string name, bool lazyCreate = true)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw new ArgumentException(nameof(name));
+
+            var pane = new OutputWindowPane(name, Guid.NewGuid());
+
+            if (!lazyCreate)
+            {
+                await pane.EnsurePaneAsync();
+            }
+
+            return pane;
+        }
+
+        /// <summary>
+        /// Gets an existing Output window pane.
+        /// Throws if a pane with the specified guid does not exist.
+        /// </summary>
+        /// <param name="guid">The pane's unique identifier.</param>
+        /// <returns>A new OutputWindowPane.</returns>
+        public static async Task<OutputWindowPane> GetAsync(Guid guid)
+        {
+            // Empty string for `_name` signals to EnsurePaneAsync that we want to get an existing pane.
+            var pane = new OutputWindowPane(string.Empty, guid);
+
+            await pane.EnsurePaneAsync();
+
+            return pane;
+        }
+
+        /// <summary>
+        /// Uniquely identifies the Output window pane.
+        /// After creating a pane, you can cache this Guid and later obtain the same pane from <see cref="GetAsync(Guid)"/>.
+        /// </summary>
+        public Guid Guid { get; }
+
+        /// <summary>
+        /// The name (title) of the Output window pane.
+        /// </summary>
+        public string Name
```

Comment:
What's the use case for changing the name of the output window pane? This seems like a bad UX practice